README.md

```diff
@@ -71,42 +79,82 @@ CREATE EXTENSION vectors;
 We support three operators to calculate the distance between two vectors:
 
 - `<->`: square Euclidean distance
-- `<#>`: dot product distance
-- `<=>`: cosine distance
+- `<#>`: negative dot product distance
+- `<=>`: negative square cosine distance
 
 ```sql
 -- call the distance function through operators
 
 -- square Euclidean distance
-SELECT array[1, 2, 3] <-> array[3, 2, 1];
+SELECT '[1, 2, 3]' <-> '[3, 2, 1]';
 -- dot product distance
-SELECT array[1, 2, 3] <#> array[3, 2, 1];
+SELECT '[1, 2, 3]' <#> '[3, 2, 1]';
 -- cosine distance
-SELECT array[1, 2, 3] <=> array[3, 2, 1];
+SELECT '[1, 2, 3]' <=> '[3, 2, 1]';
 ```
 
+Note that, "square Euclidean distance" is defined as $ \Sigma (x_i - y_i) ^ 2 $, "negative dot product distance" is defined as $ - \Sigma x_iy_i $, and "negative square cosine distance" is defined as $ - \frac{(\Sigma x_iy_i)^2}{\Sigma x_i^2 \Sigma y_i^2} $, so that you can use `ORDER BY` to perform a KNN search directly without a `DESC` keyword.
+
 ### Create a table
 
 You could use the `CREATE TABLE` statement to create a table with a vector column.
 
 ```sql
 -- create table
-CREATE TABLE items (id bigserial PRIMARY KEY, emb numeric[]);
+CREATE TABLE items (id bigserial PRIMARY KEY, emb vector(3));
 -- insert values
-INSERT INTO items (emb) VALUES (ARRAY[1,2,3]), (ARRAY[4,5,6]);
+INSERT INTO items (emb) VALUES ('[1,2,3]'), ('[4,5,6]');
 -- query the similar embeddings
-SELECT * FROM items ORDER BY emb <-> ARRAY[3,2,1]::real[] LIMIT 5;
+SELECT * FROM items ORDER BY emb <-> '[3,2,1]' LIMIT 5;
 -- query the neighbors within a certain distance
-SELECT * FROM items WHERE emb <-> ARRAY[3,2,1]::real[] < 5;
+SELECT * FROM items WHERE emb <-> '[3,2,1]' < 5;
```

Comment:
Can we still support `ARRAY[3,2,1]::real[]` like grammar? And what's the error message now if user do `SELECT * FROM items WHERE emb <-> '[a,2,1]' < 5; That the string cannot be parsed
`

Reply:
Now the message is `Bad charactor with ascii 0x61.`. Since ASCII of `a` is `0x61`.

Reply:
(For the question below.) Yes, we can recover the index from the memory-mapped file. If a structure needs to allocate space in the memory-mapped file, the root pointer of the structure is stored so that we can recover the structure from the pointer.